contracts/core/Allo.sol

```diff
@@ -1,79 +1,359 @@
+
+        address distributionStrategy =
+            _cloneDistributionStrategy ? Clone.createClone(_distributionStrategy, _nonce++) : _distributionStrategy;
+
+        return _createPool(_identityId, allocationStrategy, payable(distributionStrategy), _token, _amount, _metadata);
+    }
 
-    // Public getter on the pools mapping
-    /// @notice calls out to the registry to get the identity metadata
-    function getPoolInfo(uint256 _poolId) external view returns (PoolData memory, string memory) {
-        // Implement the function here
+    /// @notice Creates a new pool
+    /// @param _identityId The identityId of the pool creator in the registry
+    /// @param _allocationStrategy The address of the allocation strategy
+    /// @param _distributionStrategy The address of the distribution strategy
+    /// @param _token The address of the token that the pool is denominated in
+    /// @param _amount The amount of the token to be deposited into the pool
+    /// @param _metadata The metadata of the pool
+    function createPool(
+        bytes32 _identityId,
+        address _allocationStrategy,
+        address payable _distributionStrategy,
+        address _token,
+        uint256 _amount,
+        Metadata memory _metadata
+    ) external payable returns (uint256 poolId) {
+        return _createPool(_identityId, _allocationStrategy, _distributionStrategy, _token, _amount, _metadata);
     }
 
-    // @todo insert clonable strategy library, including validation that an existing pool has safe strategy
+    /// @notice Creates a new pool
+    /// @param _identityId The identityId of the pool creator in the registry
+    /// @param _allocationStrategy The address of the allocation strategy
+    /// @param _distributionStrategy The address of the distribution strategy
+    /// @param _token The address of the token that the pool is denominated in
+    /// @param _amount The amount of the token to be deposited into the pool
+    /// @param _metadata The metadata of the pool
+    function _createPool(
+        bytes32 _identityId,
+        address _allocationStrategy,
+        address payable _distributionStrategy,
+        address _token,
+        uint256 _amount,
+        Metadata memory _metadata
+    ) internal returns (uint256 poolId) {
+        if (!registry.isOwnerOrMemberOfIdentity(_identityId, msg.sender)) {
+            revert NO_ACCESS_TO_ROLE();
+        }
+
+        if (usedStrategies[_allocationStrategy] || usedStrategies[_distributionStrategy]) {
+            revert STRATEGY_ALREADY_USED();
+        }
+
+        usedStrategies[_allocationStrategy] = true;
+        usedStrategies[_distributionStrategy] = true;
+
+        Pool memory pool = Pool({
+            identityId: _identityId,
+            allocationStrategy: IAllocationStrategy(_allocationStrategy),
+            distributionStrategy: IDistributionStrategy(_distributionStrategy),
+            metadata: _metadata
+        });
+
+        poolId = _poolIndex++;
```

Comment:
This is a weird one but this will start at `poolId = 0`. This can sometimes cause issues because you might want to check `if (poolId == 0)` somewhere as proof that it doesn't exist.

I would recommend using `++poolIndex` to start at 1, or manually incrementing it in the constructor / initializer to make sure it starts at 1.